src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -692,4 +692,34 @@ export default class ReimbursementRequestService {
     if (!fileData) throw new NotFoundException('Image File', fileId);
     return fileData;
   }
+
+  /**
+   * Edits the vendor name
+   *
+   * @param name the new vendor name
+   * @param vendorId the requested vendor to be edited
+   * @param submitter the user editing the vendor name
+   * @returns the updated vendor
+   */
+  static async editVendors(name: string, vendorId: string, submitter: User) {
+    if (!isAdmin(submitter.role)) {
```

Comment:
Almost there! To properly check the user permission for the edit vendors, let's do the following:
1. use `await validateUserIsPartOfFinanceTeam(submitter);` to first check if the user is in finance team as either lead or head by using try and catch.
2. In a case when the user is not in finance team, then we just check if the user's role is at least admin. (in a catch case of try & catch). Note that if `!isAdmin(submitter.role)` is true, then we know `!isHead(submitter.role)` is true. You can take a look at the definition of these permission check utils functions.
3. Lastly, if the user is not at least an admin, then we gonna throw the AccessDenied exception, where the msg will be only the admin or finance lead or head can edit vendors.
4. Make this permission check as a helper method in RR utils file. Similar example can be found here for edit RR user permission check: https://github.com/Northeastern-Electric-Racing/FinishLine/pull/1672/files#diff-fc838d062bcc6c9ab4a0798a759ab857af3bb5e94e9ef0485d0881a9a19eef63